src/CoreServices/FSEvents.cs

```diff
@@ -199,6 +199,30 @@ public FSEventStream (string [] pathsToWatch, TimeSpan latency, FSEventStreamCre
 		{
 		}
 
+		[DllImport (Constants.CoreServicesLibrary)]
+		static extern IntPtr FSEventStreamCreateRelativeToDevice (IntPtr allocator, FSEventStreamCallback callback, 
+		IntPtr context, ulong deviceToWatch, IntPtr pathsToWatchRelativeToDevice, 
+		ulong sinceWhen, double latency, FSEventStreamCreateFlags flags);
+
+		public FSEventStream (CFAllocator allocator, ulong deviceToWatch, NSArray pathsToWatchRelativeToDevice, 
+			ulong sinceWhen, TimeSpan latency, FSEventStreamCreateFlags flags)
+		{
+			if (pathsToWatchRelativeToDevice == null) {
+				throw new ArgumentNullException ("pathsToWatchRelativeToDevice");
+			}
+
+			eventsCallback = new FSEventStreamCallback (EventsCallback);
+
+			handle =  FSEventStreamCreateRelativeToDevice (
+				allocator ==  null ? IntPtr.Zero : allocator.Handle,
+				eventsCallback, IntPtr.Zero, deviceToWatch, pathsToWatchRelativeToDevice.Handle,
+				sinceWhen, latency.TotalSeconds, flags);
+			
+			if (handle == IntPtr.Zero) {
+				throw new Exception ("Unable to create FSEventStream");
```

Comment:
This exception is too general! Use InvalidOperationException or something more precise. https://docs.microsoft.com/en-us/dotnet/api/system.invalidoperationexception?view=netframework-4.8